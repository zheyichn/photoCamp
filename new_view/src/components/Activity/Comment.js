import React from "react";
import { Avatar, Grid, Typography } from "@mui/material";

export default function Comment(props) {
  const { commentText, commentorName, commentorProfile, createdAt } = props;
  const formattedTime = new Date(createdAt).toLocaleDateString("en-US");
  return (
    <Grid
      container
      flexDirection="row"
      justifyContent="flex-start"
      alignItems="center"
      mb={2}
    >
      <Grid container item justifyContent="flex-start">
        <Grid item>
          <Avatar src={commentorProfile} alt={commentorName} />
        </Grid>
        <Grid item pl={1} pt={1}>
          <Typography data-testid="test-name" variant="subtitle1">
            {commentorName ? commentorName : "userName"}
          </Typography>
        </Grid>
        <Grid item pl={1} pt={1.5}>
          <Typography sx={{ fontSize: "14px" }}>{formattedTime}</Typography>
        </Grid>
      </Grid>

      <Grid item ml={1} mt={1}>
        <Typography>{commentText}</Typography>
      </Grid>
    </Grid>
  );
}
