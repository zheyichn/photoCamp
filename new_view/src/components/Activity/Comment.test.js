import React from "react";
import { render, screen } from "@testing-library/react";
import Comment from "./Comment";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

describe("Comment component renders correctly", () => {
  // Test1 comment display
  const testCommentorName = "Thomas";
  const testCommentText =
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec ultrices luctus tortor, in ultrices velit elementum sit amet. Lorem ipsum dolor sit amet, consectetur adipiscing elit.";
  const testCommentorProfile =
    "https://www.freepik.com/free-photo/portrait-cheerful-caucasian-man_2753777.htm#query=portrait&position=2&from_view=keyword";
  const createdAt = 1667787227;
  test("commentor name displayed correctly", () => {
    render(
      <Comment
        commentorName={testCommentorName}
        commentText={testCommentText}
        commentorProfile={testCommentorProfile}
        createdAt={createdAt}
      />
    );
    const commentorName = screen.getByTestId("test-name");
    expect(commentorName).toHaveTextContent("Thomas");
  });

  it("Comment renders correctly", () => {
    const tree = renderer
      .create(
        <Comment
          commentorName={testCommentorName}
          commentText={testCommentText}
          commentorProfile={testCommentorProfile}
          createdAt={createdAt}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
