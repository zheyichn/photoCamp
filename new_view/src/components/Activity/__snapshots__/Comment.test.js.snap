// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component renders correctly Comment renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-yto8eg-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item css-19hkhe-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
      >
        <img
          alt="Thomas"
          className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="https://www.freepik.com/free-photo/portrait-cheerful-caucasian-man_2753777.htm#query=portrait&position=2&from_view=keyword"
        />
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-1629zra-MuiGrid-root"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
        data-testid="test-name"
      >
        Thomas
      </h6>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-17apry7-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
      >
        1/20/1970
      </p>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item css-xutcuh-MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec ultrices luctus tortor, in ultrices velit elementum sit amet. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </p>
  </div>
</div>
`;
